### PR TITLE
Add GroupBy.sortedSet and sortedMap

### PR DESCRIPTION
**Pull request by timowest**
Backports for #1116 and https://github.com/querydsl/querydsl/issues/1112

### CODE REVIEW

File: querydsl-core/src/main/java/com/mysema/query/group/GroupBy.java [hunk 1]
```diff
@@ -106,43 +103,200 @@
...
+
+
+    /**
+     * Create a new aggregating map expression using a backing LinkedHashMap
      *
      * @param key
      * @param value
      * @return
      */
-    @WithBridgeMethods(value=Expression.class,castRequired=true)
```
Comment (Shredder121) [inline querydsl-core/src/main/java/com/mysema/query/group/GroupBy.java]: Are you sure the bridge method is not necessary?

Comment (timowest) [inline querydsl-core/src/main/java/com/mysema/query/group/GroupBy.java]: I will fix that

**Comment by timowest** [comment]
@Shredder121 Could you take a look at this? Is it good to merge?

**Comment by Shredder121** [comment]
Just wondering, did you actually look at the signatures?
You never answered my question.

**Comment by timowest** [comment]
Sorry for forgetting to answer. No, I didn't look. I think you posted a signature comparison already some time ago and the change seemed to be backwards compatible.

**Comment by Shredder121** [comment]
No I haven't posted a signature comparison, only a way of obtaining one.
I just [made one for this changeset](https://gist.github.com/Shredder121/a7882cda241a84ff0936), to the best of my ability (windows and all).

Anyway, most notable are [this one](https://gist.github.com/Shredder121/a7882cda241a84ff0936#file-output-diff-L61) and [this one](https://gist.github.com/Shredder121/a7882cda241a84ff0936#file-output-diff-L262), where the erased runtime method name stay the same.

One notable difference that I picked up: The numbering of inner classes is off. I am looking into it now.
If the existing inner classes come first, the numbering is fine again. (otherwise i.e. GMap$1 is not the GMap$1 that you think it is, and it is actually moved to GMap$4.

Look for example at [this](https://gist.github.com/Shredder121/a7882cda241a84ff0936#file-output-diff-L219), where the GroupCollector is moved to [GSet$4](https://gist.github.com/Shredder121/a7882cda241a84ff0936#file-output-diff-L252)

**Comment by Shredder121** [comment]
GMap is now fixed, looking into GSet.

**Comment by Shredder121** [comment]
I fixed it now, [see revisions](https://gist.github.com/Shredder121/a7882cda241a84ff0936/revisions).

Note that GSet is harder to read, since [abstract is added](https://gist.github.com/Shredder121/a7882cda241a84ff0936#file-output-diff-L275), but all new classes come [after GSet$1](https://gist.github.com/Shredder121/a7882cda241a84ff0936#file-output-diff-L243).

**Comment by Shredder121** [comment]
It probably didn't matter, but the 'diff'(of the signatures) is much cleaner now in my opinion.

**Comment by Shredder121** [comment]
Actually, I think in a serialisation sense it did matter.

**Comment by timowest** [comment]
This looks good now in my opinion.

**Comment by Shredder121** [comment]
I also think it now looks good.

**Comment by Shredder121** [comment]
I don't quite know how to improve serialisability or compatibility in serialising of anonymous inner classes, but in a members sense, I think nothing has changed in the GroupCollectors.

**Comment by timowest** [comment]
We also don't enforce backwards compatibility of the serialized form in general, so this is fine with me.

**Comment by Shredder121** [comment]
Then I'll merge this.

**Pull request merged**